From 34bcd599916852760ddb3307fe5ae5582fc5b449 Mon Sep 17 00:00:00 2001
From: Igor Grinberg <igrinberg@habana.ai>
Date: Wed, 29 May 2019 16:19:48 +0300
Subject: [PATCH] [SW-4185] preboot: let device specific load addr

This is in preparation for loading Gaudi preboot into the scratch pad
SRAM, which in turn needed as a W/A for the PLL clock switch h/w bug.

Change-Id: I86e3f422ae00a0d2dd2d12c92f4d3eeb1fb13381
Signed-off-by: Igor Grinberg <igrinberg@habana.ai>
---

diff --git a/Makefile b/Makefile
index 7d84dbc..f89df63 100644
--- a/Makefile
+++ b/Makefile
@@ -15,7 +15,7 @@
 
 SFLAGS := $(CFLAGS)
 
-LDFLAGS := -g -T common/preboot.lds -nostdlib -static
+LDFLAGS := -g -T $(HL_DEV)/preboot.lds -nostdlib -static
 LDFLAGS += -L ../../host/opt/ext-toolchain/lib/gcc/aarch64-linux-gnu/6.4.1
 LDLIBS := -lgcc
 
diff --git a/common/preboot.lds b/common/preboot.lds
deleted file mode 100644
index ea1f531..0000000
--- a/common/preboot.lds
+++ /dev/null
@@ -1,23 +0,0 @@
-/*
- * (C) Copyright 2017 HabanaLabs, Ltd.
- * Noam Camus <noamikong@gmail.com>
- *
- * SPDX-License-Identifier:     GPL-2.0+
- */
-
-MEMORY { .sram : ORIGIN = 0x7ff0000000,
-		LENGTH = 0x10000 }
-
-OUTPUT_FORMAT("elf64-littleaarch64", "elf64-littleaarch64", "elf64-littleaarch64")
-OUTPUT_ARCH(aarch64)
-ENTRY(_start)
-SECTIONS
-{
-	.text : {
-		. = ALIGN(8);
-		common/start.o (.text*)
-		* (.text*)
-	} >.sram
-
-	.note.gnu.build-id : { *(.note.gnu.build-id) } > .sram
-}
diff --git a/gaudi/preboot.lds b/gaudi/preboot.lds
new file mode 100644
index 0000000..ea1f531
--- /dev/null
+++ b/gaudi/preboot.lds
@@ -0,0 +1,23 @@
+/*
+ * (C) Copyright 2017 HabanaLabs, Ltd.
+ * Noam Camus <noamikong@gmail.com>
+ *
+ * SPDX-License-Identifier:     GPL-2.0+
+ */
+
+MEMORY { .sram : ORIGIN = 0x7ff0000000,
+		LENGTH = 0x10000 }
+
+OUTPUT_FORMAT("elf64-littleaarch64", "elf64-littleaarch64", "elf64-littleaarch64")
+OUTPUT_ARCH(aarch64)
+ENTRY(_start)
+SECTIONS
+{
+	.text : {
+		. = ALIGN(8);
+		common/start.o (.text*)
+		* (.text*)
+	} >.sram
+
+	.note.gnu.build-id : { *(.note.gnu.build-id) } > .sram
+}
diff --git a/goya/preboot.lds b/goya/preboot.lds
new file mode 100644
index 0000000..ea1f531
--- /dev/null
+++ b/goya/preboot.lds
@@ -0,0 +1,23 @@
+/*
+ * (C) Copyright 2017 HabanaLabs, Ltd.
+ * Noam Camus <noamikong@gmail.com>
+ *
+ * SPDX-License-Identifier:     GPL-2.0+
+ */
+
+MEMORY { .sram : ORIGIN = 0x7ff0000000,
+		LENGTH = 0x10000 }
+
+OUTPUT_FORMAT("elf64-littleaarch64", "elf64-littleaarch64", "elf64-littleaarch64")
+OUTPUT_ARCH(aarch64)
+ENTRY(_start)
+SECTIONS
+{
+	.text : {
+		. = ALIGN(8);
+		common/start.o (.text*)
+		* (.text*)
+	} >.sram
+
+	.note.gnu.build-id : { *(.note.gnu.build-id) } > .sram
+}
